### PR TITLE
fix(lambda): setting log retention to INFINITE causes failure

### PR DESCRIPTION
**Pull request by nija-at**
Retention period was wongly compared against NodeJs' Infinity instead
of RetentionDays.INFINITE.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
The JSDoc here should be adapted accordingly
https://github.com/aws/aws-cdk/blob/d51350b6e801750d17046ba7f09fda2bf76efbec/packages/%40aws-cdk/aws-lambda/lib/function.ts#L234-L241

File: packages/@aws-cdk/aws-lambda/test/test.log-retention.ts [hunk 1]
```diff
@@ -93,5 +93,26 @@ export = {
...
+
+    expect(stack).to(haveResource('Custom::LogRetention',
+      (resourceProps: any, inspection: InspectionFailure) => {
+        if (resourceProps.RetentionInDays !== undefined) {
+          inspection.failureReason = 'RetentionInDays was present when it was not expected';
+          return false;
+        }
+        return true;
+      }
+    ));
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda/test/test.log-retention.ts]: wasn't it possible to simply assert against the synthesized output? Why do we need this inspection callback?

Comment (nija-at) [reply packages/@aws-cdk/aws-lambda/test/test.log-retention.ts]: How do you test for the absence of a property from the synthesized output?

The template is huuge to use `matchTemplate()` while `haveResource()` can only test for the presence of AFAIK. Is there a assertion matcher or flag I'm missing?

Comment (eladb) [reply packages/@aws-cdk/aws-lambda/test/test.log-retention.ts]: Oh... maybe you can use “not(haveResource”

Comment (nija-at) [reply packages/@aws-cdk/aws-lambda/test/test.log-retention.ts]: Still doesn't quite work. 

With `not(haveResource(`, I can check that it's not a specific value but the correct assertion is that it's not any value.

The bug that this is fixing set `RetentionInDays` to 9999.

**Comment by eladb** [comment]
@nija-at you can use this now: https://github.com/aws/aws-cdk/pull/5908#pullrequestreview-346597362

**Comment by nija-at** [comment]
@eladb - Done https://github.com/aws/aws-cdk/pull/5908/commits

**Pull request merged**